python/ray/util/queue.py

```diff
@@ -212,8 +218,16 @@ def get_nowait_batch(self, num_items: int) -> List[Any]:
 
         return ray.get(self.actor.get_nowait_batch.remote(num_items))
 
+    def shutdown(self) -> None:
+        """Terminates the underlying QueueActor.
+
+        All of the resources reserved by the queue will be released.
+        """
+        if self.actor:
+            ray.kill(self.actor)
```

Comment:
Should this immediate terminate and fail pending tasks, or do we need some kind of "graceful shutdown" protocol? As-written, this is basically like sending `SIGKILL` to the queue - everything pending will fail.

I would also add `no_restart=True` to `ray.kill` here (doesn't matter because we're not setting restarts, but might in the future).

Also might be worth waiting until the actor is killed here instead of returning immediately.

Reply:
Yep good point- I added a force option to shutdown to either gracefully terminate or immediately kill.